# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.h
# Commit: ededd346172e
# Full Hash: ededd346172e646670a3e7d6e752d00ff3f77b4f
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-03-17 05:13:41
# Description:
#   Bug 1621748 - Fix stale pointers when updating history entries consecutively, r=peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66631
# ==============================================================================

diff -r e87acf21935d -r ededd346172e docshell/base/nsDocShell.h
--- a/docshell/base/nsDocShell.h	Mon Mar 16 20:29:20 2020 +0200
+++ b/docshell/base/nsDocShell.h	Mon Mar 16 18:25:45 2020 +0000
@@ -663,7 +663,9 @@
   // Call this method to swap in a new history entry to m[OL]SHE, rather than
   // setting it directly. This completes the navigation in all docshells
   // in the case of a subframe navigation.
-  void SetHistoryEntry(nsCOMPtr<nsISHEntry>* aPtr, nsISHEntry* aEntry);
+  // Returns old mOSHE/mLSHE.
+  already_AddRefed<nsISHEntry> SetHistoryEntry(nsCOMPtr<nsISHEntry>* aPtr,
+                                               nsISHEntry* aEntry);
 
   // This method calls SetHistoryEntry and updates mOSHE and mLSHE in BC to be
   // the same as in docshell
